test/unit/Makefile.am

```diff
@@ -145,11 +145,15 @@ if HAVE_PTHREADS
 check_PROGRAMS += \
     mt_a2a \
     mt_contention \
-    gettid_register \
     mt_membar \
     threading \
     web \
     thread_wait
+
+if SHMEMX_TESTS
+check_PROGRAMS += \
+    gettid_register
```

Comment:
Do we also need `pcontrol` and `shmem_ct` here?

Reply:
It's a little sloppy, but those are each handled through separate paths.  The `shmem_ct` test is effectively disabled externally by setting `USE_PORTALS4` to `false` in the tests-sos `configure.ac`.  The `pcontrol` test is enabled/disabled (disabled by default) with a separate `--enable-profiling` option that sets the AM conditional `ENABLE_PROFILING`.